packages/snap/src/currency.ts

```diff
@@ -151,8 +151,8 @@ export async function getOptimalFeeCurrency(
   const gasLimit = (await wallet.estimateGas(tx)).mul(5);
   const celoBalance = await wallet.getBalance();
   const tokenAddresses = await feeCurrencyWhitelistContract.getWhitelist();
-
-  if (gasLimit.add(tx.value ?? 0) >= celoBalance) {
+  const gasLimitPlusValue = gasLimit.add(tx.value ?? 0);
+  if (gasLimitPlusValue.gt(celoBalance)) {
```

Comment:
gasLimit is denomited in gas whereas  value is denomiced in CELO. 

you need to multiply gasLimit by GasPrice to get the amount of CELO that will be spent.

Reply:
before adding to value